pkg/host/machine_info_linux.go

```diff
@@ -125,9 +125,34 @@ func readKVMInfo(buffer *bytes.Buffer) error {
 	return nil
 }
 
+func getModuleTextAddr(moduleName string) (uint64, error) {
+	addrPath := filepath.Join("/sys", "module", moduleName, "sections", ".text")
+	addrContent, err := os.ReadFile(addrPath)
+	if err != nil {
+		return 0, fmt.Errorf("could not read module .text address file: %w", err)
+	}
+	addrString := strings.TrimSpace(string(addrContent))
+	addr, err := strconv.ParseUint(addrString, 0, 64)
+	if err != nil {
+		return 0, fmt.Errorf("address parsing error in %v: %w", moduleName, err)
+	}
+	return addr, nil
+}
+
 func getModulesInfo() ([]KernelModule, error) {
 	modulesText, _ := os.ReadFile("/proc/modules")
-	return parseModules(modulesText)
+	modules, err := parseModules(modulesText)
+	if err != nil {
+		return modules, err
+	}
+	// Fix up module addresses to use .text addresses where available.
+	for i, module := range modules {
+		addr, err := getModuleTextAddr(module.Name)
+		if err == nil {
```

Comment:
They do happen for several modules. We could instead stop the error propagation and return 0 from getModuleTextAddr() - would you prefer that?

Reply:
> They do happen for several modules. 

What were the reasons? Are they bad enough to abort the whole process?

Reply:
The reason was that for certain modules this file did not exist. I don't think we need to abort the process, because  these modules may not have coverage at all. Falling back to /proc/modules plus getModuleTextAddr() sounds sane.

Reply:
Did you have a chance to see why the file does not exist -- is it normal?

It would be great to be able to notice if/when something suddenly breaks. If we just silently fall back to the other approach, it won't be possible.

Reply:
I figured out the offending modules have zero .text size.
So in fact these modules aren't needed by syzkaller and can be deleted. Let me try that.

Reply:
Done, PTAL